File: 백준/Silver/5800. 성적 통계/성적 통계.py
import sys
input = sys.stdin.readline

K = int(input())
classes = []

for _ in range(K):
    classes.append(list(map(int, input().split()))[1:])
    classes[-1].sort(reverse=True)

for i in range(1, K+1):
    print(f"Class {i}")
    max_value = max(classes[i-1])
    min_value = min(classes[i-1])
    largest_gap = 0
    length = len(classes[i-1])

    for j in range(length-1):
        largest_gap = max(largest_gap, classes[i-1][j] - classes[i-1][j+1])

    print(f"Max {max_value}, Min {min_value}, Largest gap {largest_gap}")

File: 백준/Silver/5800. 성적 통계/성적 통계.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main()  {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int K;
    cin >> K;

    for (int i = 1; i <= K; ++i) {
        int N;
        cin >> N;
        int* scores = new int[N];

        for (int j = 0; j < N; ++j) {
            cin >> scores[j];
        }

        sort(scores, scores + N, greater<>());

        int maxScore = scores[0];
        int minScore = scores[N-1];
        int maxDiff = 0;

        for (int j = 0; j < N-1; ++j) {
            int diff = scores[j] - scores[j+1];
            maxDiff = diff > maxDiff ? diff : maxDiff;
        }

        cout << "Class " << i << '\n';
        cout << "Max " << maxScore << ", Min " << minScore << ", Largest gap " << maxDiff << '\n';
        delete[] scores;
    }
}